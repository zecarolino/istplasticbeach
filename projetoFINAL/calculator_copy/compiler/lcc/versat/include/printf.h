#ifndef _PRINTF_H_
#define _PRINTF_H_

#include "printi.h"
#include "puts.h"
#include "stdarg.h"

int printf(const char *fmt, ...)
{
  va_list ap;
  va_start(ap, fmt);
  while (*fmt != 0) {
    if (*fmt == '%') {
      fmt++;
      if (*fmt == 'd') printi(va_arg(ap,int), 10);
      else if (*fmt == 's') puts(va_arg(ap,char*));
      else if (*fmt == 'c') putchar(va_arg(ap,int));
      else putchar(*fmt);
    }
    else putchar(*fmt);
    fmt++;
  }
  va_end(ap);
  return 1;
}

#endif /* _PRINTF_H_ */
