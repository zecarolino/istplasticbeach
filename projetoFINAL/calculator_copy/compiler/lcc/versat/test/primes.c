int nextPrime(int n)
{
  int i;

  if (n <= 0) return 0;
  if (n <= 2) return n+1;
  while (++n > 0) {
    if (n % 2 == 0) continue; /* is even */
    for (i = 3; i < n; i += 2) /* test odd numbers */
      if (n % i == 0)
	break;
    if (i >= n)
      return n;
  }
  return 0; /* overflow */
}

#include "printf.h"
int main()
{
  int max = 20, n = 1, cnt = 0;

  while (n < max && n > 0)
    printf("%d: %d\n", ++cnt, n = nextPrime(n));

  return 0;
}
