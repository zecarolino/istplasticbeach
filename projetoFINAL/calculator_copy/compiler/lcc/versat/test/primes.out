VCD info: dumpfile xtop.vcd opened for output.
1: 2
2: 3
3: 5
4: 7
5: 11
6: 13
7: 17
8: 19
9: 23
0
Catched Trap at time  75452 clock cycles:
